frontend/src/components/scan/utils/RecentScans.jsx

```diff
@@ -94,39 +91,60 @@ RecentScansCard.defaultProps = {
 };
 
 export default function RecentScans({ classification, param }) {
-  // user recent scans
-  const [recentScansUser, LoaderRecentScansUser, refetch] =
-    useAxiosComponentLoader({
-      url: JOB_RECENT_SCANS_USER,
-      method: "POST",
-    });
-  console.debug("recentScansUser", recentScansUser);
+  // api
+  const [
+    loadingScansUser,
+    loadingScans,
+    recentScansUserError,
+    recentScansError,
+    recentScansUser,
+    recentScans,
+    fetchRecentScansUser,
+    fetchRecentscans,
+  ] = useRecentScansStore((state) => [
+    state.loadingScansUser,
+    state.loadingScans,
+    state.recentScansUserError,
+    state.recentScansError,
+    state.recentScansUser,
+    state.recentScans,
+    state.fetchRecentScansUser,
+    state.fetchRecentscans,
+  ]);
 
-  React.useEffect(() => {
-    refetch();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  console.debug(
+    "loadingScansUser",
+    loadingScansUser,
+    "loadingScans",
+    loadingScans,
+  );
 
   // file md5
   const [fileMd5, setFileMd5] = React.useState("");
   if (classification === "file" && param) {
     param.text().then((x) => setFileMd5(md5(x)));
   }
-  // observable/file recent scans
-  const [recentScans, LoaderRecentScans] = useAxiosComponentLoader({
-    url: JOB_RECENT_SCANS,
-    method: "POST",
-    data: { md5: fileMd5.length ? fileMd5 : md5(param) },
-  });
+
+  React.useEffect(() => {
+    fetchRecentScansUser();
+    fetchRecentscans(fileMd5.length ? fileMd5 : md5(param));
+  }, [fetchRecentScansUser, fetchRecentscans, fileMd5, param]);
```

Comment:
I suggest to split this useEffect in two: we will have one useEffect only for the intial fetching (user) and another use effect only for the observable. In this way we will reduce the number of requests to the backend. I think there shouldn't be drawbacks: in case you start a scan and come back to the scan page the initial request should be done the same. What do you think ?

Reply:
yep that is very important to save network and backend computation